ONSdigital/rm-sdx-gateway
# Expose receipting endpoint via HTTPS

# Motivation and Context

Enabling HTTPS for receipting from SDX to RAS/RM. Unfortunately this needs to be done by the application rather than a load balancer as its an internal component.

# What has changed



1. Configured Spring Boot to start a HTTPs connection on a new port.
1. Add a new connector for HTTP on the old port to maintain backwards compatibility. We'll eventually turn this off.
1. Added self sign cert for local testing and use in dev environments only. Preprod and Prod will have certificates provisioned securely.

# How to test?



1. Deploy into a kubernetes cluster
2. Port forward to both the http and https ports and ensure you can hit the endpoints.

# Links



https://trello.com/c/f5ATkLBL/2002-s10a-add-port-443-and-self-signed-cert-to-rm-sdx-gateway-8
